File: Python/Sort and Search/BubbleSort.py
from nose.tools import assert_equal

def bubbleSort(arr):
	for i in range(len(arr)):
		for j in range(i+1, len(arr)):
			if arr[j] < arr[i]:
				arr[i], arr[j] = arr[j], arr[i]
	return arr

class Bubble_Sort(object):
	def test(self, func):
		assert_equal(func([1, 7, 4, 2, 10, 8, 20, 3]), [1, 2, 3, 4, 7, 8, 10, 20])
		assert_equal(func([1, 2, 3, 4, 5]), [1, 2, 3, 4, 5])
		print('TESTS PASSED')

t = Bubble_Sort()
t.test(bubbleSort)

File: Sort and Search/BubbleSort.js
function bubbleSort(array){
	for(var i=0;i<array.length;i++){
		for(var j=1;j<array.length-i;j++){
			if(array[j]<array[j-1]){
				temp = array[j];
				array[j] = array[j-1];
				array[j-1] = temp;
			}
		}
	}
	return array;
}

console.log(bubbleSort([3, 7, 4, 1, 9, 10]));
console.log(bubbleSort([1, 2, 3, 4, 5, 6]));
console.log(bubbleSort([10, 4, 1, 9, 2, 7, 8]));